ios/browser/api/sync/brave_sync_worker.cc

```diff
@@ -260,32 +260,6 @@ bool BraveSyncWorker::IsFirstSetupComplete() {
 										sync_service->GetUserSettings()->IsFirstSetupComplete();
 }
 
-bool BraveSyncWorker::RemoveDevice(const std::string& device_guid) {
```

Comment:
Since https://github.com/brave/brave-core/pull/6732 (Remove other devices from the sync chain) has been merged , we can bring back the function but we need to integrate with `brave_sync::DeleteDevice` also check if `is_self_delete_supported` of a remote device before showing the delete option. Self reset when receiving remote delete message is handled in `BraveProfileSyncService` so we don't need to do anything but if ios observes SyncState to do the similar thing, we should remove that and let `BraveProfileSyncService` does the work.

cc @AlexeyBarabash in case I miss any details